### PR TITLE
CLOUDP-47064: add license

### PR DESCRIPTION
**Pull request by jamesbroadhead**

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+      To apply the Apache License to your work, attach the following
+      boilerplate notice, with the fields enclosed by brackets "[]"
+      replaced with your own identifying information. (Don't include
+      the brackets!)  The text should be enclosed in the appropriate
+      comment syntax for the file format. We also recommend that a
+      file or class name and description of purpose be included on the
+      same "printed page" as the copyright notice for easier
+      identification within third-party archives.
+
+   Copyright [yyyy] [name of copyright owner]
```
Comment (gssbzn) [inline LICENSE]: This was merged already but this line should be properly attributed, 
2019 MongoDB, Inc.

Comment (fabianlindfors) [reply LICENSE]: Is that not directions for how to use the Apache license? Perhaps the appendix should not be included at all.

Comment (gssbzn) [reply LICENSE]: Ha, I was mostly use to MIT which you actually update this file but Apache you leave this as is and add a `NOTICE` file with this copy corrected, and/or this copy on each source file... so scratch my comment and some copy to a bunch of files to be done

Comment (fabianlindfors) [reply LICENSE]: I see! Thanks for keeping an eye out :)

Comment (jamesbroadhead) [reply LICENSE]: My reading of https://www.apache.org/dev/licensing-howto.html is that we would have to include a NOTICE file if we were an Apache project -- but as it stands, all we need to do is inject MongoDB & the year into this file. 

@fabianlindfors could you do that please?

Comment (fabianlindfors) [reply LICENSE]: I'm sorry, I'm not too familiar with licenses in general. Do we need to add `Copyright 2019 MongoDB, Inc` to `LICENSE` or to all source files?

Comment (jamesbroadhead) [reply LICENSE]: leave it to me :)

Comment (fabianlindfors) [reply LICENSE]: Might be for the best 😅

**Pull request merged**